FreeRTOS/coreHTTP
# WIP: Add Chunked Body Reading Support

Description
-----------
This PR introduces a new function, `HTTPClient_Read`, to coreHTTP to enable reading the response body in chunks. This enhancement addresses the requirement captured in issue #160, facilitating memory-efficient handling of large responses.

Test Steps
-----------
1. Checkout the branch containing the changes.
2. Build coreHTTP with the new updates.
3. Run existing tests to ensure no regressions.
4. Execute new unit tests covering the `HTTPClient_Read` functionality.
5. Manually test `HTTPClient_Read` with varying response sizes and chunk sizes.

Checklist:
----------
- [ ] Tested the changes. No regression in existing tests.
- [ ] Modified and/or added unit-tests covering the code changes in this PR.

Related Issue
-----------
This PR addresses issue #160.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
